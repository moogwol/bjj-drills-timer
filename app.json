{
  "expo": {
    "name": "bjj-drills-timer",
    "slug": "bjj-drills-timer",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/bjj-patch.png",
    "scheme": "myapp",
    "userInterfaceStyle": "automatic",
    "splash": {
      "image": "./assets/images/bjj-patch.png",
      "resizeMode": "contain",
      "backgroundColor": "#f8f2da"
    },
    "ios": {
      "supportsTablet": true
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/images/bjj-patch-icon.png",
        "backgroundColor": "#ffffff"
      },
      "package": "com.doodlesack.bjjdrillstimer"
    },
    "web": {
      "bundler": "metro",
      "output": "static",
      "favicon": "./assets/images/favicon.png"
    },
    "plugins": [
      "expo-router"
    ],
    "experiments": {
      "typedRoutes": true
    },
    "extra": {
      "router": {
        "origin": false
      },
      "eas": {
        "projectId": "c207b331-95d4-4d0e-9519-cc3abe2287b4"
      }
    },
    "runtimeVersion": {
      "policy": "appVersion"
    },
    "updates": {
      "url": "https://u.expo.dev/c207b331-95d4-4d0e-9519-cc3abe2287b4"
    }
  }
}
