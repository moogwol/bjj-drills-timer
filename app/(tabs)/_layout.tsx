import { Tabs } from "expo-router";

export default function RootLayout() {
  return (
    <Tabs screenOptions={{tabBarActiveTintColor: 'blue'}} >
      <Tabs.Screen
       name="index"
       options={{
          tabBarLabel: "Home",
          headerTitle: "Home",        
       }}
       />
      <Tabs.Screen name="workoutScreen"
      options={{
        tabBarLabel: "Workout",
        headerTitle: "Workout",
      }} />
    </Tabs>
  );
}
