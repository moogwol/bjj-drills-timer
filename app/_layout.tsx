import { Stack } from "expo-router/stack";
import { createContext, useState } from "react";
import { PaperProvider, MD2LightTheme, MD3Colors, MD2DarkTheme, MD3LightTheme } from "react-native-paper";


export interface Drill {
  id: number;
  name: string;
}

import React from 'react';

interface DrillContextType {
  drills: Drill[];
  setDrills: React.Dispatch<React.SetStateAction<Drill[]>>;
}

// Initialize the context with undefined. The actual value will be provided by a Provider.


export const DrillContext = React.createContext<DrillContextType | undefined>(undefined);

export default function Layout() {
  const [drills, setDrills] = useState<Drill[]>([]);


  return (
    <PaperProvider theme={MD3LightTheme}>
      <DrillContext.Provider value={{ drills, setDrills }}>
        <Stack>
          <Stack.Screen name="(tabs)"
            options={{ headerShown: false }}
          />
        </Stack>
      </DrillContext.Provider>
    </PaperProvider>
  );
}