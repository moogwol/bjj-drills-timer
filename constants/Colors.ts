/**
 * Below are the colors that are used in the app. The colors are defined in the light and dark mode.
 * There are many other ways to style your app. For example, [Nativewind](https://www.nativewind.dev/), [Tamagui](https://tamagui.dev/), [unistyles](https://reactnativeunistyles.vercel.app), etc.
 */


// blue theme
const blueTheme = {
  primary: "#4341AB",
  secondary: "white",
  accent: "#231FEA",
  light: "#EDC81A",
  dark: "#1F1F31",
};

export const useColours = () => {
  return blueTheme;
};