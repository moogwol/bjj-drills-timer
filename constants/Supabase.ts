import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = "https://xflgmbxgfafdodkvsiss.supabase.co";
const PUBLIC_ANON_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InhmbGdtYnhnZmFmZG9ka3ZzaXNzIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTg1MzY5NjQsImV4cCI6MjAzNDExMjk2NH0.HO2AIC8c6x3LPANor04JInW2ivqV_4a8lDDlRBl38GE"
 
export const supabase = createClient(SUPABASE_URL, PUBLIC_ANON_KEY);

