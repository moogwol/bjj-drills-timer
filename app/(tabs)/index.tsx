import { View, StyleSheet } from "react-native";
import { useContext, useState, useEffect } from "react";
import { List, Checkbox, Button } from "react-native-paper";
import { useColours } from "@/constants/Colors";
import { fetchDrillsByTags } from "@/libraries/http";
import { Drill, DrillContext } from "@/app/_layout";
import { router } from "expo-router";
import { NumberPicker } from "@/components/NumberPicker";

const colours = useColours();


export default function Index() {

  const [numDrills, setNumDrills] = useState(1);
  const drillContext = useContext(DrillContext);
  const [workouButtonDisabled, setWorkoutButtonDisabled] = useState(true);

  // Checkbox states
  const [judoChecked, setJudoChecked] = useState(false);
  const [basicChecked, setBasicChecked] = useState(false);
  const [advancedChecked, setAdvancedChecked] = useState(false);
  const [takedownsChecked, setTakedownsChecked] = useState(false);
  const [roundsOnlyChecked, setRoundsOnlyChecked] = useState(false);


  // Check if the context is defined
  if (!drillContext) {
    throw new Error("DrillContext is not defined");
  }

  const { drills, setDrills } = drillContext;


  // Handle the click event for the "Go to workout" button  
  const handleClickGoToWorkout = async () => {
    if (!roundsOnlyChecked) {
      // Get the drills from the database
      const tags = [];
      if (judoChecked) tags.push('judo');
      if (basicChecked) tags.push('basic');
      if (advancedChecked) tags.push('advanced');
      if (takedownsChecked) tags.push('takedown');
      console.log('Chosen tags:', tags);
      const data = await fetchDrillsByTags(tags, numDrills);
      setDrills(data as Drill[]);
      console.log(data);
      router.push('/workoutScreen')
    } else {
      // Create an array of round with the round number
      const drills = []
      for (let i = 0; i < numDrills; i++) {
        drills.push({ name: 'Round ' + (i + 1) });
        router.push('/workoutScreen')
      }
      setDrills(drills as Drill[]);
    }
  }

  const handleClickDecrementDrills = async () => {
    numDrills > 1 && setNumDrills(numDrills - 1);
  };

  const handleClickIncrementDrills = async () => {
    setNumDrills(numDrills + 1);
  };

  // Set the workout button to disabled if no checkboxes are checked
  useEffect(() => {
    if (roundsOnlyChecked || judoChecked || basicChecked || advancedChecked || takedownsChecked) {
      setWorkoutButtonDisabled(false);
    } else {
      setWorkoutButtonDisabled(true);
    }
  }, [roundsOnlyChecked, judoChecked, basicChecked, advancedChecked, takedownsChecked]);  


  return (
    <View style={styles.container}>
      <NumberPicker title={`Number of ${roundsOnlyChecked ? 'rounds' : 'drills'}`}
        decrement={handleClickDecrementDrills}
        increment={handleClickIncrementDrills}
        num={numDrills}
      />

      <List.Section style={styles.accordionSection}>
        <List.Accordion testID="accordion" title="Choose your drills" titleStyle={styles.accordion} style={styles.accordion} >
          <Checkbox.Item testID="roundsCheckbox" label="Rounds only" labelStyle={styles.checkboxLabel} color={colours.light} uncheckedColor={colours.light} status={roundsOnlyChecked ? 'checked' : 'unchecked'} onPress={() => setRoundsOnlyChecked(!roundsOnlyChecked)} />
          <Checkbox.Item label="Judo" labelStyle={styles.checkboxLabel} color={colours.light} uncheckedColor={colours.light} status={judoChecked ? 'checked' : 'unchecked'} disabled={roundsOnlyChecked} onPress={() => setJudoChecked(!judoChecked)} />
          <Checkbox.Item label="Basic" labelStyle={styles.checkboxLabel} color={colours.light} uncheckedColor={colours.light} status={basicChecked ? 'checked' : 'unchecked'} disabled={roundsOnlyChecked} onPress={() => setBasicChecked(!basicChecked)} />
          <Checkbox.Item label="Advanced" labelStyle={styles.checkboxLabel} color={colours.light} uncheckedColor={colours.light} status={advancedChecked ? 'checked' : 'unchecked'} disabled={roundsOnlyChecked} onPress={() => setAdvancedChecked(!advancedChecked)} />
          <Checkbox.Item label="Takedowns" labelStyle={styles.checkboxLabel} color={colours.light} uncheckedColor={colours.light} status={takedownsChecked ? 'checked' : 'unchecked'} disabled={roundsOnlyChecked} onPress={() => setTakedownsChecked(!takedownsChecked)} />
        </List.Accordion>
      </List.Section>
      <Button testID="workoutButton" disabled={workouButtonDisabled} buttonColor={colours.primary} mode="contained" onPress={handleClickGoToWorkout}>Go to workout</Button>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: colours.secondary,
    color: colours.light,
    padding: 10,
  },
  accordionSection: {
    width: "100%",
    backgroundColor: colours.accent,
    color: colours.light,
  },
  accordion: {
    width: "100%",
    backgroundColor: colours.accent,
    color: colours.light,
  },
  listItem: {
    backgroundColor: colours.accent,
  },
  listItemTitle: {
    color: colours.light,
  },
  text: {

  },
  numberPicker: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
  },
  checkbox: {
    backgroundColor: colours.accent,
    fontSize: 50,
    fontWeight: "bold",
  },
  checkboxLabel: {
    color: colours.light,
  },
});