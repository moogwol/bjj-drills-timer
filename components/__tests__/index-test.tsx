import Index from "@/app/(tabs)";
import '@testing-library/jest-native/extend-expect';
import { Drill, DrillContext } from "@/app/_layout";
import { useColours } from "@/constants/Colors";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

// Mock go to workout function
const mockGoToWorkout = jest.fn();


// Check that the componment renders correctly
const colours = useColours();
describe('Index', () => { }); {
    it('checks index screen renders correctly', async () => {
        const { toJSON } = render(
        <DrillContext.Provider value={{ drills: [], setDrills: () => {} }}>
            <Index />
        </DrillContext.Provider>);
        expect(toJSON()).toMatchSnapshot();
        });
}

// Test the handleClickGoToWorkout function does not work before any checkboxes are checked
describe('Index', () => {
    it('calls handleClickGoToWorkout function when Go to workout button is pressed', async () => {
        const { getByTestId } = render(
            <DrillContext.Provider value={{ drills: [], setDrills: () => {} }}>
                <Index />
            </DrillContext.Provider>
        );

        fireEvent.press(getByTestId('workoutButton'));
        expect(mockGoToWorkout).not.toHaveBeenCalled();
    });
});

// // Test the handleClickGoToWorkout function works after a checkbox has been checked
// describe('Index', () => {
//     it('calls handleClickGoToWorkout function when Go to workout button is pressed', async () => {
//         const { getByTestId } = render(
//             <DrillContext.Provider value={{ drills: [], setDrills: () => {} }}>
//                 <Index />
//             </DrillContext.Provider>
//         );

//         fireEvent.press(getByTestId('accordion'));
//         fireEvent.press(getByTestId('roundsCheckbox'));
//         fireEvent.press(getByTestId('workoutButton'));
//         expect(mockGoToWorkout).toHaveBeenCalled();
//     });
// });
