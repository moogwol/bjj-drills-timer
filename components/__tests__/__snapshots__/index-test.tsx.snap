// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks index screen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "color": "#EDC81A",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 24,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 32,
            },
            {
              "margin": 10,
            },
          ],
        ]
      }
    >
      Number of drills
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 23,
            "height": 46,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 46,
          }
        }
        testID="decrement-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 23,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="decrement-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 23,
                  },
                ],
              ]
            }
            testID="decrement"
          >
            <Text />
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 36,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 44,
              },
              undefined,
            ],
          ]
        }
      >
        1
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="increment-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="increment-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 20,
                  },
                ],
              ]
            }
            testID="increment"
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginVertical": 8,
        },
        {
          "backgroundColor": "#231FEA",
          "color": "#EDC81A",
          "width": "100%",
        },
      ]
    }
    theme={
      {
        "animation": {
          "scale": 1,
        },
        "colors": {
          "backdrop": "rgba(50, 47, 55, 0.4)",
          "background": "rgba(255, 251, 254, 1)",
          "elevation": {
            "level0": "transparent",
            "level1": "rgb(247, 243, 249)",
            "level2": "rgb(243, 237, 246)",
            "level3": "rgb(238, 232, 244)",
            "level4": "rgb(236, 230, 243)",
            "level5": "rgb(233, 227, 241)",
          },
          "error": "rgba(179, 38, 30, 1)",
          "errorContainer": "rgba(249, 222, 220, 1)",
          "inverseOnSurface": "rgba(244, 239, 244, 1)",
          "inversePrimary": "rgba(208, 188, 255, 1)",
          "inverseSurface": "rgba(49, 48, 51, 1)",
          "onBackground": "rgba(28, 27, 31, 1)",
          "onError": "rgba(255, 255, 255, 1)",
          "onErrorContainer": "rgba(65, 14, 11, 1)",
          "onPrimary": "rgba(255, 255, 255, 1)",
          "onPrimaryContainer": "rgba(33, 0, 93, 1)",
          "onSecondary": "rgba(255, 255, 255, 1)",
          "onSecondaryContainer": "rgba(29, 25, 43, 1)",
          "onSurface": "rgba(28, 27, 31, 1)",
          "onSurfaceDisabled": "rgba(28, 27, 31, 0.38)",
          "onSurfaceVariant": "rgba(73, 69, 79, 1)",
          "onTertiary": "rgba(255, 255, 255, 1)",
          "onTertiaryContainer": "rgba(49, 17, 29, 1)",
          "outline": "rgba(121, 116, 126, 1)",
          "outlineVariant": "rgba(202, 196, 208, 1)",
          "primary": "rgba(103, 80, 164, 1)",
          "primaryContainer": "rgba(234, 221, 255, 1)",
          "scrim": "rgba(0, 0, 0, 1)",
          "secondary": "rgba(98, 91, 113, 1)",
          "secondaryContainer": "rgba(232, 222, 248, 1)",
          "shadow": "rgba(0, 0, 0, 1)",
          "surface": "rgba(255, 251, 254, 1)",
          "surfaceDisabled": "rgba(28, 27, 31, 0.12)",
          "surfaceVariant": "rgba(231, 224, 236, 1)",
          "tertiary": "rgba(125, 82, 96, 1)",
          "tertiaryContainer": "rgba(255, 216, 228, 1)",
        },
        "dark": false,
        "fonts": {
          "bodyLarge": {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          "bodyMedium": {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          "bodySmall": {
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "400",
            "letterSpacing": 0.4,
            "lineHeight": 16,
          },
          "default": {
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          "displayLarge": {
            "fontFamily": "System",
            "fontSize": 57,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 64,
          },
          "displayMedium": {
            "fontFamily": "System",
            "fontSize": 45,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 52,
          },
          "displaySmall": {
            "fontFamily": "System",
            "fontSize": 36,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 44,
          },
          "headlineLarge": {
            "fontFamily": "System",
            "fontSize": 32,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 40,
          },
          "headlineMedium": {
            "fontFamily": "System",
            "fontSize": 28,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 36,
          },
          "headlineSmall": {
            "fontFamily": "System",
            "fontSize": 24,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 32,
          },
          "labelLarge": {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
          },
          "labelMedium": {
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 16,
          },
          "labelSmall": {
            "fontFamily": "System",
            "fontSize": 11,
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 16,
          },
          "titleLarge": {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          "titleMedium": {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          "titleSmall": {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
          },
        },
        "isV3": true,
        "roundness": 4,
        "version": 3,
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": false,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "paddingRight": 24,
                  "paddingVertical": 8,
                },
                {
                  "backgroundColor": "#231FEA",
                  "color": "#EDC81A",
                  "width": "100%",
                },
              ],
            ]
          }
          testID="accordion"
        >
          <View
            pointerEvents="none"
            style={
              {
                "flexDirection": "row",
                "marginVertical": 6,
              }
            }
          >
            <View
              style={
                [
                  {
                    "paddingLeft": 16,
                  },
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      {
                        "backgroundColor": "#231FEA",
                        "color": "#EDC81A",
                        "width": "100%",
                      },
                    ],
                  ]
                }
              >
                Choose your drills
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(28, 27, 31, 0.12)",
        "borderRadius": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="workoutButton-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(28, 27, 31, 0.12)",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="workoutButton-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="workoutButton"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ]
            }
            testID="workoutButton-text"
          >
            Go to workout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
